File: sdk/python/hitechcloud_tools/resources/qr_and_thanh_toan.py
"""QR & Thanh toán resource (5 endpoints)"""
from .base import BaseResource

class QrAndThanhToanResource(BaseResource):
    """QR & Thanh toán"""

    def banks(self, **params) -> dict:
        """GET /api/banks - Danh bạ ngân hàng Việt Nam (65 ngân hàng): mã..."""
        return self._http.get("/api/banks", params or None)

    def banks_2(self, **params) -> dict:
        """GET /api/banks/2 - Tra một ngân hàng theo mã BIN (970436)"""
        return self._http.get("/api/banks/2", params or None)

    def generator_vietqr(self, **params) -> dict:
        """GET /api/generator/vietqr - Sinh mã VietQR theo chuẩn NAPAS để nhận chuyển..."""
        return self._http.get("/api/generator/vietqr", params or None)

    def tools_qr_generate(self, **data) -> dict:
        """POST /api/tools/qr/generate - Sinh mã QR từ nội dung bất kỳ"""
        return self._http.post("/api/tools/qr/generate", data or None)

    def tools_qr_vcard(self, **data) -> dict:
        """POST /api/tools/qr/vcard - Sinh mã QR danh thiếp vCard"""
        return self._http.post("/api/tools/qr/vcard", data or None)

File: sdk/ruby/lib/hitechcloud_tools/resources/qr_and_thanh_toan.rb
# QR & Thanh toán resource (5 endpoints)
class QrAndThanhToanResource
  def initialize(client)
    @client = client
  end

  # GET /api/banks - Danh bạ ngân hàng Việt Nam (65 ngân hàng): mã...
  def banks(params = {})
    @client.get("/api/banks", params)
  end

  # GET /api/banks/2 - Tra một ngân hàng theo mã BIN (970436)
  def banks_2(params = {})
    @client.get("/api/banks/2", params)
  end

  # GET /api/generator/vietqr - Sinh mã VietQR theo chuẩn NAPAS để nhận chuyển...
  def generator_vietqr(params = {})
    @client.get("/api/generator/vietqr", params)
  end

  # POST /api/tools/qr/generate - Sinh mã QR từ nội dung bất kỳ
  def tools_qr_generate(data = {})
    @client.post("/api/tools/qr/generate", data)
  end

  # POST /api/tools/qr/vcard - Sinh mã QR danh thiếp vCard
  def tools_qr_vcard(data = {})
    @client.post("/api/tools/qr/vcard", data)
  end

end
